src/Context/ContextDefinition.php

```diff
@@ -78,6 +79,21 @@ public static function createFromArray($values) {
     if (isset($values['class']) && !in_array('Drupal\rules\Context\ContextDefinitionInterface', class_implements($values['class']))) {
       throw new \Exception('ContextDefinition class must implement \Drupal\rules\Context\ContextDefinitionInterface.');
     }
+
+    // Annotation classes extract data from passed annotation classes directly
+    // used in the classes they pass to.
+    foreach (['label', 'description'] as $key) {
+      // @todo Remove this workaround in https://www.drupal.org/node/2362727.
+      if (isset($values[$key])) {
+        if ($values[$key] instanceof TranslatableMarkup) {
+          $values[$key] = (string) $values[$key];
```

Comment:
@claudiu-cristea we should instead test for `instanceof \Drupal\Core\Annotation\Translation`.  See https://www.drupal.org/node/2610458 for a test that demostrates the problem as well.